enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/statemachine/service/StateMachineService.java

```diff
@@ -126,7 +126,7 @@ public Message<OnboardingEvent> createMessage(OwnerId ownerId, String processId,
     @Transactional(readOnly = true)
     public void changeMachineStatesInBatch() {
         final AtomicInteger countFinished = new AtomicInteger(0);
-        try (Stream<IdentityVerificationEntity> stream = identityVerificationService.streamAllIdentityVerificationsToChangeState().parallel()) {
+        try (Stream<IdentityVerificationEntity> stream = identityVerificationService.streamAllIdentityVerificationsToChangeState()) {
               stream.filter(identityVerification -> identityVerification.getDocumentVerifications().stream()
```

Comment:
🤔 The root cause is most probably lazy-loading in the filter. Shouldn't we fetch it instead?

Reply:
Let's rewrite the subselect.